node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+use crate::sub_lib::wallet::Wallet;
+use itertools::Either;
+use masq_lib::logger::Logger;
+use std::collections::HashMap;
+use std::fmt::{Display, Formatter};
+use std::time::SystemTime;
+use thousands::Separable;
+use variant_count::VariantCount;
+use web3::types::U256;
+use masq_lib::utils::convert_collection;
+
+// PaymentAdjuster is a very efficient recursive and scalable algorithm that inspects payments under
+// the condition of an acute insolvency. You can expand the scope of the evaluation by writing your
+// own CriterionCalculator, that should be specialized on a distinct parameter of a payable account,
+// and sticking it inside the vector that stores them.
+
+pub type AdjustmentAnalysisResult =
+    Result<Either<Vec<QualifiedPayableAccount>, AdjustmentAnalysis>, PaymentAdjusterError>;
+
+pub trait PaymentAdjuster {
+    fn search_for_indispensable_adjustment(
+        &self,
+        qualified_payables: Vec<QualifiedPayableAccount>,
+        agent: &dyn BlockchainAgent,
+    ) -> AdjustmentAnalysisResult;
+
+    fn adjust_payments(
+        &mut self,
+        setup: PreparedAdjustment,
+        now: SystemTime,
+    ) -> Result<OutboundPaymentsInstructions, PaymentAdjusterError>;
+
+    as_any_ref_in_trait!();
+}
+
+pub struct PaymentAdjusterReal {
+    analyzer: PreparatoryAnalyzer,
+    disqualification_arbiter: DisqualificationArbiter,
+    service_fee_adjuster: Box<dyn ServiceFeeAdjuster>,
+    calculators: Vec<Box<dyn CriterionCalculator>>,
+    inner: Box<dyn PaymentAdjusterInner>,
+    logger: Logger,
+}
+
+impl PaymentAdjuster for PaymentAdjusterReal {
+    fn search_for_indispensable_adjustment(
+        &self,
+        qualified_payables: Vec<QualifiedPayableAccount>,
+        agent: &dyn BlockchainAgent,
+    ) -> AdjustmentAnalysisResult {
+        let disqualification_arbiter = &self.disqualification_arbiter;
+        let logger = &self.logger;
+
+        self.analyzer
+            .analyze_accounts(agent, disqualification_arbiter, qualified_payables, logger)
+    }
+
+    fn adjust_payments(
+        &mut self,
+        setup: PreparedAdjustment,
+        now: SystemTime,
+    ) -> Result<OutboundPaymentsInstructions, PaymentAdjusterError> {
+        let analyzed_payables = setup.adjustment_analysis.accounts;
+        let response_skeleton_opt = setup.response_skeleton_opt;
+        let agent = setup.agent;
+        let initial_service_fee_balance_minor = agent.service_fee_balance_minor();
+        let required_adjustment = setup.adjustment_analysis.adjustment;
+        let largest_exceeding_balance_recently_qualified =
+            find_largest_exceeding_balance(&analyzed_payables);
+
+        self.initialize_inner(
+            initial_service_fee_balance_minor,
+            required_adjustment,
+            largest_exceeding_balance_recently_qualified,
+            now,
+        );
+
+        let sketched_debug_info_opt = self.sketch_debug_info_opt(&analyzed_payables);
```

Comment:
We don't want to confuse `info` with the log level here, so prefer a rename `sketch_debug_log_opt`. We want to do the same on line 133 with the function `complete_debug_info_if_enabled()`.

Also, don't forget to rename the variable name.